### ISSUE TITLE
【FR】Bug: FlagGems full/ones/zeros operators cause 21% accuracy degradation on microsoft/Phi-4-mini-instruct

### ISSUE THREAD

**Issue by ckxud** [FlagOpen/FlagGems#4893]
## Summary

When evaluating **microsoft/Phi-4-mini-instruct** on NVIDIA H20-3e for FlagOS migration, V2 (FlagGems fully enabled, 24 operators) scores **30.0%** on GPQA Diamond vs NV baseline **38.0%**, a relative degradation of **21.05%**, far exceeding the 5% threshold.

After operator-level accuracy tuning, **disabling the tensor-construction operator group `{full, ones, zeros}` restores the score to 42.0%** (relative to NV baseline: -10.53%, i.e. better than baseline → passes). This localizes the problematic operators to FlagGems' tensor-construction ops `full` / `ones` / `zeros`.

## Environment

| Item | Value |
|------|-------|
| Hardware | nvidia |
| GPU | NVIDIA H20-3e ×8 |
| PyTorch | 2.11.0+cu130 |
| FlagGems | 5.0.2 |
| vLLM | 0.20.2 |
| Model | microsoft/Phi-4-mini-instruct |
| Benchmark | GPQA Diamond (50 questions, greedy) |

## Affected Operators

| Operator | Category | Evidence |
|----------|----------|----------|
| `full` | tensor construction | Disabling the group restored 30.0% → 42.0% |
| `ones` | tensor construction | Same (cumulative group disable) |
| `zeros` | tensor construction | Same (cumulative group disable) |

Operators kept after tuning (18-op whitelist, excluding the three above plus performance-pruned add/arange_start/cos/exponential_/general_mm/index):
`argmax, cat, copy_, lt_scalar, pow_scalar, rand_like, reciprocal, scatter_, sin, softmax, softmax_out, sub, to_copy, true_divide, true_divide_, where_self, where_self_out, zero_`

## Steps to Reproduce

1. Launch container on nvidia (vLLM 0.20.2 + FlagGems 5.0.2)
2. Start vLLM serve with `USE_FLAGGEMS=1`, all 24 operators in whitelist
3. Run GPQA Diamond (50 questions, greedy) → score 30.0%
4. NV baseline (same model/benchmark, FlagGems disabled) → 38.0%, relative degradation 21.05% > 5%
5. Disable operators group by group: after disabling `{full, ones, zeros}`, score recovers to 42.0%

## Expected Behavior

With FlagGems fully enabled, relative accuracy degradation should be ≤ 5%. The tensor-construction operators `full/ones/zeros` should not introduce a 21% overall accuracy degradation.

## Actual Behavior

- V2 full: 30.0% (21.05% degradation)
- Disabling `{full, ones, zeros}`: 42.0% (passes)

## Tuning Log

```
[2026-07-17 15:22:09] V2 | accuracy below threshold (NV baseline mode, no V1)
  detail: V2=30.0%, NV baseline=38.0%, relative degradation=21.05% (threshold 5%)
  action: mark accuracy_ok=false, trigger step-5 operator accuracy tuning
[2026-07-17 15:30:46] V2 | accuracy tuning disabled operators
  detail: after disabling {full, ones, zeros}, V2=42.0% (NV baseline 38.0%, relative -10.53%, passes)
  result: accuracy_ok=true
```

## Suggested Investigation

1. Unit-test the numeric/dtype/device semantics of `full` / `ones` / `zeros` against the PyTorch reference (especially fill-value dtype promotion, default device handling).
2. Check whether these operators produce wrong initial values in Phi-4-mini's (Phi3ForCausalLM) mask / position_ids construction path (e.g. uninitialized memory, wrong dtype causing mask to be ineffective).
3. Since these are tensor-construction (not compute) operators, focus on whether their output feeds into the subsequent attention/causal mask; a wrong initial value there would strongly amplify accuracy degradation.
4. Re-verify with a larger evaluation set (full 198 questions) to rule out the statistical noise of the 50-question run (σ≈6.5%).
